actionpack/lib/action_controller/metal/etag_with_template_digest.rb

```diff
@@ -25,7 +25,7 @@ module EtagWithTemplateDigest
       class_attribute :etag_with_template_digest
       self.etag_with_template_digest = true
 
-      ActiveSupport.on_load :action_view, yield: true do
+      ActiveSupport.on_load :action_controller, yield: true do
```

Comment:
We need to change this completely. We only want to execute this when action view is included and when the action_controller is loaded. In order to match both the first requirement we need to move this code to the action view railtie.